### ISSUE TITLE
Accessing nonexistent properties is not checked in JS files

### ISSUE THREAD

**Issue by que-etc** [Microsoft/TypeScript#22288]
**TypeScript Version:** 2.8.0-dev.20180302

**Search Terms:**
* checkJs
* noImplicitAny
* object
* undefined

**Code**

```javascript
const fruits = {
    PINEAPPLE: 'tasty'
};

// Name of the property is mistyped and running this code in JS engines
// will result in a TypeError.
fruits.PINAPPLE.toUpperCase();
```

**Expected behavior:**
Following error should be raised if `checkJs` and `noImplicitAny` flags are enabled:
`Property 'PINAPPLE' does not exist on type '{ PINEAPPLE: string; }'. Did you mean 'PINEAPPLE'?`

**Actual behavior:**
No error is shown.

**Related Issues:** #16137 - this one describes a similar problem, but for some reason it was closed by reporter.

**Comment by andy-ms** [comment]
Probably a bug since JS objects created by object literals allow assignment of new properties -- but a property access should still be an error.

**Comment by mhegazy** [comment]
Objects in a .js file are assumed to be open-ended. in other words they behave as if they had ` { [x:string]: any; }` on them.
